### PR TITLE
Rallpack1: Add z_min as a membrane

### PR DESCRIPTION
**Pull request by jlallouette**
This PR is intended for debugging an issue in which the addition of a second membrane to rallpack1 leads to a difference in the results between STEPS 3 and STEPS 4.
It could be related to the caburst issue.

### CODE REVIEW

**Comment by jlallouette** [comment]
download
download

File: rallpack1/rallpack1.py [hunk 1]
```diff
@@ -139,9 +140,9 @@ def run(seed, mesh_path, steps_version):
     # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
     sim.newRun()
 
-    sim.TRIS(membrane.tris).Leak[leaksus].Count = 1
+    sim.TRIS(memb.tris).Leak[leaksus].Count = 1
```
Comment (cattabiani) [inline rallpack1/rallpack1.py:143]: are we leaking in the same way? Just to be sure

Comment (jlallouette) [reply rallpack1/rallpack1.py:143]: Yes, I modified it otherwise STEPS 3 would also have leak on z_min

Comment (jlallouette) [reply rallpack1/rallpack1.py:143]: Update: I applied this change to the 1 membrane script just to be sure. It gives the same plots as above.

**Comment by iahepburn** [comment]
is this with the bugfixed STEPS 4 code, Jules?

**Comment by jlallouette** [comment]
Yes, it is with `katta/fix_inputs_v_i_on_ghosts`

File: rallpack1/rallpack1.py (hunk 1 above)
Comment (iahepburn) [reply rallpack1/rallpack1.py:143]: not sure if this is causing the issue, but there's a problem in line 90 isn't there? The capacitance of both 'membrane' and 'membrane2' are corrected by the same factor, which comes from comparing the analytical surface area of cylinder with one face to the mesh one. But I guess this will not give the desired effect, which is that the overall capacitance of the two membranes match the ideal capacitance from the curved surface of a cylinder and the face of a cylinder respectively. 
Long story short, I think  we need two core_fac_areas applied separately.

Comment (iahepburn) [reply rallpack1/rallpack1.py:143]: for STEPS3 it'd have to be done per triangle (because there aren't separate membranes), though I'm not sure if this is actually causing the issue- the way it is now looks like it's consistent between STEPS 3 and 4 so probably not.

File: rallpack1/rallpack1.py [hunk 2]
```diff
@@ -139,9 +140,9 @@ def run(seed, mesh_path, steps_version):
...
 
-    sim.membrane.Potential = -65e-3
+    sim.ALL(Membrane).Potential = -65e-3
```
Comment (iahepburn) [inline rallpack1/rallpack1.py:144]: what happens internally when this line is run? In EField every node has a voltage value so every one has to be set to the same potential. There's only one membrane in STEPS 3 so there can be no ambiguity, all nodes are internal to the one membrane. What happens in STEPS 4 if there are multiple membranes?

File: rallpack1/rallpack1.py (hunk 1 above)
Comment (jlallouette) [reply rallpack1/rallpack1.py:142]: I updated the script to remove the correction factors, since they are only relevant for comparing to the analytical trace.
I still get the same plots as above.

Comment (iahepburn) [reply rallpack1/rallpack1.py:142]: are all nodes in the EField solution at -65mV at the start of the simulation? First thing to check.

File: rallpack1/rallpack1.py (hunk 2 above)
Comment (jlallouette) [reply rallpack1/rallpack1.py:144]: It sets the voltage of all nodes in the mesh twice. Once for `membrane`, and once again for `membrane2`.

File: rallpack1/rallpack1.py (hunk 1 above)
Comment (iahepburn) [reply rallpack1/rallpack1.py:142]: and which one is actually moving? Presumably STEPS 4 moves down to ~-65.3mV though we're only seeing the relative figures so STEPS 3 could be moving up to ~-64.7mV.

**Comment by jlallouette** [comment]
STEPS 3 goes up when `z_min` is added as a membrane:
download

download

**Comment by iahepburn** [comment]
oh- weird. So STEPS 3 has the issue perhaps. Both are moving actually but STEPS 3 moves more. Impossible to say which one actually has the issue here without investigating further. 

We have, fundamentally, Q=CV here. We increase C (by adding more membrane), Q is constant which means to my mind V should decrease a bit (relative to the 1 membrane model). So I'm confused.

File: rallpack1/rallpack1.py (hunk 1 above)
Comment (jlallouette) [reply rallpack1/rallpack1.py:142]: Regarding the potential of nodes at t=0, they are all at -65mV:
download

**Comment by jlallouette** [comment]
Without leak, STEPS 3 and STEPS 4 are nearly identical:

download
download

**Comment by jlallouette** [comment]
Unitary leak conductance was slightly lower for STEPS 3 with 2 membranes. After fixing this STEPS 3 and STEPS 4 are identical.
I'm closing the PR.
